by-language/csharp-npgsql/DemoTypes.cs

```diff
@@ -0,0 +1,188 @@
+    {
+
+        public static async Task<DataTable> AllTypesExample(NpgsqlConnection conn)
+        {
+            Console.WriteLine("Running AllTypesExample");
+
+            // Submit DDL, create database schema.
+            await using (var cmd = new NpgsqlCommand("DROP TABLE IF EXISTS testdrive.example", conn))
+            {
+                cmd.ExecuteNonQuery();
+            }
+
+            await using (var cmd = new NpgsqlCommand(@"
+    CREATE TABLE testdrive.example (
+        -- Numeric types
+        null_integer INT,
+        integer INT,
+        bigint BIGINT,
+        float FLOAT,
+        double DOUBLE,
+        decimal DECIMAL(8, 2),
+        -- Other scalar types
+        bit BIT(8),
+        bool BOOLEAN,
+        text TEXT,
+        char CHARACTER(5),
+        timestamp_tz TIMESTAMP WITH TIME ZONE,
+        timestamp_notz TIMESTAMP WITHOUT TIME ZONE,
+        ip IP,
+        -- Container types
+        ""array"" ARRAY(STRING),
+        ""object"" OBJECT(DYNAMIC),
+        -- Geospatial types
+        geopoint GEO_POINT,
+        geoshape GEO_SHAPE,
+        -- Vector type
+        ""float_vector"" FLOAT_VECTOR(3)
+    );
+            ", conn))
+            {
+                cmd.ExecuteNonQuery();
+            }
+
+            // Insert single data point.
+            await using (var cmd = new NpgsqlCommand(@"
+    INSERT INTO testdrive.example (
+        null_integer,
+        integer,
+        bigint,
+        float,
+        double,
+        decimal,
+        bit,
+        bool,
+        text,
+        char,
+        timestamp_tz,
+        timestamp_notz,
+        ip,
+        ""array"",
+        -- ""object"",
+        geopoint,
+        -- geoshape,
+        float_vector
+    ) VALUES (
+        @null_integer,
+        @integer,
+        @bigint,
+        @float,
+        @double,
+        @decimal,
+        @bit,
+        @bool,
+        @text,
+        @char,
+        @timestamp_tz,
+        @timestamp_notz,
+        @ip,
+        @array,
+        -- @object,
+        @geopoint,
+        -- @egoshape,
+        @float_vector
+    );
+", conn))
+            {
+                Console.WriteLine(cmd);
+                cmd.Parameters.AddWithValue("null_integer", DBNull.Value);
+                cmd.Parameters.AddWithValue("integer", 42);
+                cmd.Parameters.AddWithValue("bigint", 42);
+                cmd.Parameters.AddWithValue("float", 42.42);
+                cmd.Parameters.AddWithValue("double", 42.42);
+                cmd.Parameters.AddWithValue("decimal", 42.42);
+                cmd.Parameters.AddWithValue("bit", "01010101");
+                cmd.Parameters.AddWithValue("bool", true);
+                cmd.Parameters.AddWithValue("text", "foobar");
+                cmd.Parameters.AddWithValue("char", "foo");
+                cmd.Parameters.AddWithValue("timestamp_tz", "1970-01-02T00:00:00+01:00");
+                cmd.Parameters.AddWithValue("timestamp_notz", "1970-01-02T00:00:00");
```

Comment:
I am not sure about those, because apparently, they work in this context, but it is also just a very basic test anyway.
Let me reference this report here, to have a complete list of evalations here, and to reduce the surface of unknown unknowns.

- https://github.com/crate/crate-npgsql/issues/67